typing/includecore.ml

```diff
@@ -136,7 +136,7 @@ type type_mismatch =
   | Field_missing of bool * Ident.t
   | Record_representation of bool   (* true means second one is unboxed float *)
   | Unboxed_representation of bool  (* true means second one is unboxed *)
-  | Immediate
+  | Immediate of immediacy * immediacy
```

Comment:
I see your point, but at the same time I tried to make the code of this PR not specific to the current three levels of immediateness so that it's easier to add more levels in the future.

I adapted the printing code to be more generic as well.

Reply:
I would personally still prefer to narrow down the types. `immediacy * immediacy` produces 9 combinations, where only 2 are actually used. It always startles me to come back to some code and discover unused data combinations.

Also, I feel like the introduction of `immediacy_violation` (or perhaps `immediacy_mismatch`) would be very much inline with the recent work in #8828.

However, the current solution is fine as well.

Reply:
Alright, I'm giving this idea a try

Reply:
I did that in the last commit. The result looks better indeed, what do you think?